packages/mediapipe-core/lib/src/ffi_utils.dart

```diff
@@ -6,73 +6,131 @@ import 'dart:ffi';
+  /// Creates a Dart view on a pointer to utf8 chars in native memory, cast as
+  /// unsigned, 8-bit integers. This method does not copy the data out of native
+  /// memory.
+  ///
+  /// If this is called on a `nullptr`, the method returns `null`.
+  Uint8List? toUint8List(int length) {
+    if (isNullPointer) return null;
+    RangeError.checkNotNegative(length, 'length');
+    return cast<Uint8>().asTypedList(length);
+  }
+
+  /// Converts the C memory representation of a string into a Dart [String].
+  ///
+  /// If this is called on a `nullptr`, the method returns `null`.
+  ///
+  /// See also:
+  ///  * [toDartStrings]
+  String? toDartString({int? length}) {
+    if (isNullPointer) return null;
+    return cast<Utf8>().toDartString(length: length);
   }
-  return dartStrings;
 }
 
-/// Converts Dart's representation for binary data, a [Uint8List], into its C
-/// memory representation.
-Pointer<Char> prepareUint8List(Uint8List ints) {
-  final Pointer<Uint8> ptr = calloc<Uint8>(ints.length);
-  ptr.asTypedList(ints.length).setAll(0, ints);
-  return ptr.cast();
+/// Helpers to convert between `Pointer<Pointer<Char>>` and `List<String>`.
+extension DartAwarePointerChars on Pointer<Pointer<Char>> {
+  /// Creates a Dart view on a pointer to a pointer to utf8 chars in native memory,
+  /// cast as a `List<String>`. This method does not copy the data out of native
+  /// memory.
+  ///
+  /// If this is called on a `nullptr`, the method returns `null`.
+  ///
+  /// See also:
+  ///  * [toDartString], for a non-list equivalent.
+  List<String?>? toDartStrings(int length) {
+    if (isNullPointer) return null;
+    final dartStrings = <String?>[];
+    int counter = 0;
+    while (counter < length) {
+      dartStrings.add(this[counter].toDartString());
+      counter++;
+    }
+    return dartStrings;
+  }
 }
 
-/// Converts a pointer to binary data in C memory to Dart's representation for
-/// binary data, a [Uint8List].
-Uint8List toUint8List(Pointer<Char> val, {int? length}) {
-  final codeUnits = val.cast<Uint8>();
-  if (length != null) {
+/// Helpers to convert a [Pointer<Char>] into a [Uint8List].
+extension DartAwareFloats on Pointer<Float> {
+  /// Creates a Dart view on a pointer to floats in native memory, cast as a
+  /// Float32List. This method does not copy the data out of native memory.
+  ///
+  /// If this is called on a `nullptr`, the method returns `null`.
+  Float32List? toFloat32List(int length) {
+    if (isNullPointer) return null;
     RangeError.checkNotNegative(length, 'length');
-  } else {
-    length = _length(codeUnits);
+    return cast<Float>().asTypedList(length);
+  }
+}
+
+/// Extension method for copying a [Float32List] from Dart memory into native
+/// memory.
+extension NativeFloats on Float32List {
+  /// Copies a [Float32List] into native memory as a `float*`.
+  ///
+  /// Returns an [allocator]-allocated pointer to the result.
+  Pointer<Float> toNative({Allocator allocator = malloc}) {
```

Comment:
I'd probably call this `copyToNative`.

And yes maybe this should be a PR to `package:ffi`. Let's first use it here and then later upstream some useful abstractions once we have proven them here.